sound/soc/sof/ipc4-loader.c

```diff
@@ -193,6 +194,22 @@ static int sof_ipc4_query_fw_configuration(struct snd_sof_dev *sdev)
 			trace_sof_ipc4_fw_config(sdev, "Trace log size", *tuple->value);
 			ipc4_data->mtrace_log_bytes = *tuple->value;
 			break;
+		case SOF_IPC4_FW_CFG_MAX_LIBS_COUNT:
+		{
+			size_t size = sizeof(struct snd_sof_module_library_info);
+
+			ipc4_data->max_fw_libs = *tuple->value;
+
+			if (!ipc4_data->max_fw_libs)
```

Comment:
need to check max_fw_libs < N since a invalid value may use up memory.

Reply:
and how do we estimate that N @RanderWang ? Are you suggesting we limit the count in the kernel even if the FW max count is larger?

Reply:
If you grep the firmware code for MAX_LIBS and MAX_LIB_MODULES you'll get the answer that N==16

That said, existing drivers trust all the configuration information reported by firmware. Maybe this is a result of the firmware signature by Intel.

Reply:
on my TGL , the max libs supported is 32. I can cap it to 32 in the kernel